meshio/permas_io.py

```diff
@@ -132,59 +154,81 @@ def write(
-            elif n == 6:
-                if k == 0:
-                    fh.write('!\n')
-                    fh.write('        $ELEMENT TYPE = PENTA6 ESET = PENTA6\n')
-                fh.write(form % (
-                    k+1, c[0]+1, c[1]+1, c[2]+1, c[3]+1, c[4]+1, c[5]+1
-                    ))
-            elif n == 8:
-                if k == 0:
-                    fh.write('!\n')
-                    fh.write('        $ELEMENT TYPE = HEXE8 ESET = HEXE8\n')
-                fh.write(form % (
-                    k+1, c[0]+1, c[1]+1, c[2]+1, c[3]+1,
-                    c[4]+1, c[5]+1, c[6]+1, c[7]+1
-                    ))
+        #
+        # Avoid non-unique element numbers in case of multiple element types by num_ele !!!
+        #
+        num_ele = 0
+        
+        for meshio_type, cell in cells.iteritems():
+            numcells, num_local_nodes = cell.shape
+            permas_type = meshio_to_permas_type[meshio_type]
+            if permas_type != 'NA':
+                fh.write('!\n')
+                fh.write('        $ELEMENT TYPE = %s ESET = %s\n' % (permas_type, permas_type))
+                for k, c in enumerate(cell):
+                    form = '        %8d ' + ' '.join(num_local_nodes * ['%8d']) + '\n'
+                    fh.write(form % ((k+num_ele+1,) + tuple(c + 1)))
+                num_ele += numcells    
+                
         fh.write('!\n')
         fh.write('    $END STRUCTURE\n')
         fh.write('!\n')
+        elem_3D = ['HEXE8', 'TET4', 'PENTA6', 'PYRA5']
+        elem_2D = ['TRIA3', 'QUAD4']
+        fh.write('    $SYSTEM NAME = NSV\n')
+        fh.write('!\n')
+        fh.write('        $ELPROP\n')
+        for meshio_type, cell in cells.iteritems():
+            permas_type = meshio_to_permas_type[meshio_type]
+            if permas_type in elem_3D:
+                fh.write('            %s MATERIAL = STEEL\n' %permas_type)
+            elif permas_type in elem_2D:
+                fh.write('            %s GEODAT = GD_%s MATERIAL = STEEL\n' %(permas_type,permas_type))
+            else:
+                pass
+        fh.write('!\n')
+        fh.write('        $GEODAT SHELL  CONT = THICK  NODES = ALL\n')
+        for meshio_type, cell in cells.iteritems():
+            permas_type = meshio_to_permas_type[meshio_type]
+            if permas_type in elem_2D:
+                fh.write('            GD_%s 1.0\n' %permas_type)
+        fh.write('!\n')
+        fh.write('    $END SYSTEM\n')
+        fh.write('!\n')
+        fh.write('    $CONSTRAINTS NAME = SPCVAR_1\n')
+        fh.write('    $END CONSTRAINTS\n')
+        fh.write('!\n')
+        fh.write('    $LOADING NAME = LOADVAR_1\n')
+        fh.write('    $END LOADING\n')
+        fh.write('!\n')
         fh.write('$EXIT COMPONENT\n')
         fh.write('!\n')
+        fh.write('$ENTER MATERIAL\n')
+        fh.write('!\n')
+        fh.write('    $MATERIAL NAME = STEEL TYPE = ISO\n')
```

Comment:
No. I have added those lines in order to run the PERMAS Job directly. At least one material definition is needed.  Element sets are assigned to materials within the system variant.